packages/@aws-cdk/aws-vpcv2-alpha/lib/vpc-v2.ts

```diff
@@ -0,0 +1,478 @@
+
+  /**
+   * Isolated Subnets that are part of this VPC.
+   */
+  public readonly isolatedSubnets: ISubnet[];
+
+  /**
+   * Public Subnets that are part of this VPC.
+   */
+  public readonly publicSubnets: ISubnet[];
+
+  /**
+   * Pbulic Subnets that are part of this VPC.
+   */
+  public readonly privateSubnets: ISubnet[];
+
+  /**
+   * To define dependency on internet connectivity
+   */
+  public readonly internetConnectivityEstablished: IDependable;
+
+  /**
+   * To define dependency on internet connectivity
+   */
+  private readonly _internetConnectivityEstablished = new DependencyGroup();
+
+  /**
+ * reference to all secondary blocks attached
+ */
+  public readonly secondaryCidrBlock = new Array<CfnVPCCidrBlock>;
+
+  /**
+   * For validation to define IPv6 subnets, set to true in case of
+   * Amazon Provided IPv6 cidr range
+   * IPv6 addresses can be attached to the subnets
+   * @default false
+   */
+  public readonly useIpv6: boolean = false;
+
+  public readonly ipv4CidrBlock: string = '';
+
+  constructor(scope: Construct, id: string, props: VpcV2Props = {}) {
+    super(scope, id, {
+      physicalName: props.vpcName ?? Lazy.string({
+        produce: () => Names.uniqueResourceName(this, { maxLength: 128, allowedSpecialCharacters: '_' }),
+      }),
+    });
+
+    this.ipAddresses = props.primaryAddressBlock ?? IpAddresses.ipv4('10.0.0.0/16');
+    const vpcOptions = this.ipAddresses.allocateVpcCidr();
+
+    this.dnsHostnamesEnabled = props.enableDnsHostnames == null ? true : props.enableDnsHostnames;
+    this.dnsSupportEnabled = props.enableDnsSupport == null ? true : props.enableDnsSupport;
+    const instanceTenancy = props.defaultInstanceTenancy || 'default';
+    this.resource = new CfnVPC(this, 'Resource', {
+      cidrBlock: vpcOptions.ipv4CidrBlock, //for Ipv4 addresses CIDR block
+      enableDnsHostnames: this.dnsHostnamesEnabled,
+      enableDnsSupport: this.dnsSupportEnabled,
+      ipv4IpamPoolId: vpcOptions.ipv4IpamPool?.ipamPoolId, // for Ipv4 ipam option
+      ipv4NetmaskLength: vpcOptions.ipv4NetmaskLength, // for Ipv4 ipam option
+      instanceTenancy: instanceTenancy,
+    });
+
+    this.node.defaultChild = this.resource;
+    this.vpcCidrBlock = this.resource.attrCidrBlock;
+    if (vpcOptions.ipv4CidrBlock) {
+      this.ipv4CidrBlock = vpcOptions.ipv4CidrBlock;
+    }
+    this.ipv6CidrBlocks = this.resource.attrIpv6CidrBlocks;
+    this.vpcId = this.resource.attrVpcId;
+    this.vpcArn = Arn.format({
+      service: 'ec2',
+      resource: 'vpc',
+      resourceName: this.vpcId,
+    }, this.stack);
+
+    if (props.secondaryAddressBlocks) {
+      const secondaryAddressBlocks: IIpAddresses[] = props.secondaryAddressBlocks;
+
+      let ipCount = 0;
+      for (const secondaryAddressBlock of secondaryAddressBlocks) {
+
+        //Counter to generate a random string for input to hash function
+        ipCount+=1;
+        const hash = pathHash('Secondary'+ipCount);
```

Comment:
[blocking] I still do not think this is a good way, using the count in the name still will cause changes if customer changes the order or add a new block.

Reply:
The problem is everytime we send this value to pathHash it needs to be unique string, so we need something as a variable with Secondary that will not change in the next run but will be unique in list of secondary IP addresses. eg.
If we try to do this as 
const hash = pathHash('Secondary'+randomString()) ... it will keep on changing every time we do a cdk synth or cdk deploy, so we somehow need to make it a constant string but with a unique value for pathHash() function in terms of multiple secondary IPaddresses.